core/cache/redisBackend.go

```diff
@@ -0,0 +1,378 @@
+}
+
+func finalizer(b *RedisBackend) {
+	b.Close()
+}
+
+// DefaultRedisBackendOptions gets the default options for redis backend
+func DefaultRedisBackendOptions() RedisBackendOptions {
+	return RedisBackendOptions{
+		Network:     "tcp",
+		Host:        "127.0.0.1",
+		Port:        "6379",
+		Db:          0,
+		Password:    "",
+		MaxIdle:     8,
+		IdleTimeout: time.Minute * 30,
+		WriteLock:   false,
+	}
+}
+
+// NewRedisBackend creates an redis cache backend
+func NewRedisBackend(options RedisBackendOptions, logger flamingo.Logger) Backend {
+	err := mergo.Merge(&options, DefaultRedisBackendOptions())
+	if err != nil {
+		logger.WithField("category", "redisBackend").Error(fmt.Sprintf("Error merging options: %v", err))
+	}
+
+	b := &RedisBackend{
+		Pool: &redis.Pool{
+			MaxIdle:     options.MaxIdle,
+			IdleTimeout: options.IdleTimeout,
+			TestOnBorrow: func(c redis.Conn, t time.Time) error {
+				_, err := c.Do("PING")
+				return err
+			},
+			Dial: func() (redis.Conn, error) {
+				return redisConnector(options.Network, fmt.Sprintf("%v:%v", options.Host, options.Port), options.Password)
+			},
+		},
+		logger: logger,
+	}
+	b.withBackendType("redis")
+
+	runtime.SetFinalizer(b, finalizer) // close all connections on destruction
+
+	return b
+}
+
+// Inject RedisBackend dependencies
+func (b *RedisBackend) Inject(logger flamingo.Logger) {
+	b.logger = logger
+}
+
+// Close ensures all redis connections are closed
+func (b *RedisBackend) Close() {
+	b.Pool.Close()
+}
+
+// CreatePrefixedKey creates an redis-compatible key
+func (b *RedisBackend) CreatePrefixedKey(key string, prefix string) string {
+	key = redisKeyRegex.ReplaceAllString(key, "-")
+	return fmt.Sprintf("%v%v", prefix, key)
+}
+
+// Get an cache key
+func (b *RedisBackend) Get(key string) (entry *Entry, found bool) {
+	conn := b.Pool.Get()
+	defer conn.Close()
+
+	reply, err := conn.Do("GET", b.CreatePrefixedKey(key, valuePrefix))
+	if err != nil {
+		b.countMiss()
+		b.logger.WithField("category", "redisBackend").Error(fmt.Sprintf("Error getting key '%v': %v", key, err))
+		return nil, false
+	}
+	if reply == nil {
+		b.countMiss()
+		b.logger.WithField("category", "redisBackend").Error(fmt.Sprintf("Returned nil for key: %v", key))
+		return nil, false
+	}
+
+	value, err := redis.Bytes(reply, err)
+	if err != nil {
+		b.countMiss()
+		b.logger.WithField("category", "redisBackend").Error(fmt.Sprintf("Error convert value to bytes of key '%v': %v", key, err))
+		return nil, false
+	}
+
+	redisEntry, err := b.decodeEntry(value)
+	if err != nil {
+		b.countMiss()
+		b.logger.WithField("category", "redisBackend").Error(fmt.Sprintf("Error decoding content of key '%v': %v", key, err))
+		return nil, false
+	}
+
+	b.countHit()
+	return b.buildResult(redisEntry), true
+}
+
+// Set an cache key
+func (b *RedisBackend) Set(key string, entry *Entry) error {
+	conn := b.Pool.Get()
+	defer conn.Close()
+
+	if b.writeLockEnabled {
+		err := b.lock(conn, key)
+		if err != nil {
```

Comment:
unlock should be in a defer call rigth the next line

Early exists in case of errors below would not free the lock

Reply:
changed